File: frontend/src/interfaces/order.ts
import {IProfile} from "./profile";

export interface IOrder {
    id: number,
    profile: IProfile,
    status: number,
    product: {
        id: number,
        name: string,
        photo: string,
        description: string,
        price: number,
        size: number,
        amount: number,
    },
    amount: number,
    in_cart: boolean,
    total: number,
}

File: backend/src/api/internal/serializers/order.py
from rest_framework import serializers

from api.internal.models.order import Order
from api.internal.models.transaction import Transaction
from api.internal.serializers.product import ProductDetailsSerializer
from api.internal.serializers.profile import ProfileSerializer
from api.internal.services.cart import get_total


class OrderSerializer(serializers.ModelSerializer):
    profile = ProfileSerializer()
    product = ProductDetailsSerializer(source="storage_cell")
    total = serializers.SerializerMethodField(method_name="get_total")

    def get_total(self, order: Order) -> int:
        return get_total(order)

    class Meta:
        model = Order
        fields = ["id", "profile", "status", "product", "amount", "in_cart", "total"]


class FormedOrderSerializer(serializers.ModelSerializer):
    order = OrderSerializer()

    class Meta:
        model = Transaction
        fields = ["id", "created_at", "order"]
